File: pg-ephemeral/tests/integration/ruby/test_connection.rb
#!/usr/bin/env ruby

require 'pg'

# DATABASE_URL should be set by pg-ephemeral run-env
database_url = ENV.fetch('DATABASE_URL')

puts "Connecting to: #{database_url}"

# Connect using libpq-style DATABASE_URL
# The pg gem natively supports libpq connection strings
conn = PG.connect(database_url)

# Test the connection
conn.exec("SELECT 1")

puts "SUCCESS: Connected to PostgreSQL successfully"
conn.close

File: pg-ephemeral/tests/integration/prisma/test_connection.js
import { PrismaClient } from '@prisma/client';

// DATABASE_URL should be set by pg-ephemeral run-env
const databaseUrl = process.env.DATABASE_URL;

if (!databaseUrl) {
  console.error('ERROR: DATABASE_URL environment variable is not set');
  process.exit(1);
}

console.log(`Connecting to: ${databaseUrl}`);

const prisma = new PrismaClient();

try {
  // Test the connection with a simple query
  await prisma.$queryRaw`SELECT 1`;

  console.log('SUCCESS: Connected to PostgreSQL successfully');

  await prisma.$disconnect();
} catch (error) {
  console.error('ERROR: Failed to connect to PostgreSQL:', error);
  await prisma.$disconnect();
  process.exit(1);
}
